src/renderers/webgl/WebGLAttributes.js

```diff
@@ -141,11 +152,30 @@ function WebGLAttributes( gl ) {
 
 	}
 
+	function unref( geometry, attribute ) {
+
+		var refs = attributeReferences.get( geometry );
+
+		if ( refs ) {
+
+			refs.delete( attribute );
+
+		}
+
+		if ( refs.size === 0 ) {
+
```

Comment:
I am missing something or is this the wrong way round?
refs.size is the number of attributes per geometry, when you want it the other way round?

If so, you could add a container to the 'buffer' object returned from createBuffer() rather than a renderer global and maintain the set of geometries in that container?

Apart from that +1 for ability to share buffer attributes safely.

Reply:
Um yup, it's backwards. 😅

Reply:
If you pass geometry into createBuffer() you can move the overhead of grabbing the reference to creation only, so any performance hit will be negligible (in the common case just an extra parameter passed to update() is the only overhead.

Would a simple refCount property on buffer {} object be safe.  I can't see a problem with that?

Reply:
You're right, I should be using the buffer object.

I can also do a refcount, but I'd also need to track a 'last seen state' of the attributes for each geometry so that I can tell if a reference is new. I'll do that.